app/src/test/java/ch/hikemate/app/model/route/HikeTest.kt

```diff
@@ -165,4 +165,103 @@ class HikeTest {
     assertEquals(elevationGain, detailed.elevationGain, deltaForDoubleComparison)
     assertEquals(difficulty, detailed.difficulty)
   }
+
+  @Test
+  fun segmentsWorksAsExpected() {
+    // Case 1: Empty waypoints list
+    val emptyHike =
+        createHike(
```

Comment:
Praise: thank you for using the existing `createHike` function and keeping the code clean this way!